dql/parser.go

```diff
@@ -1721,8 +1721,15 @@ L:
 							itemInFunc.Errorf("len function only allowed inside inequality" +
 								" function")
 					}
-					function.Attr = nestedFunc.NeedsVar[0].Name
-					function.IsLenVar = true
+					if len(function.Attr) == 0 {
+						// eq(len(a), 1)
+						function.Attr = nestedFunc.NeedsVar[0].Name
+						function.IsLenVar = true
+					} else {
+						// eq(1, len(a))
+						return nil, itemInFunc.Errorf("incorrect order of value and len" +
```

Comment:
we can improve the error message. Do we support only eq function here? Couldn't it be gt or something too?

Reply:
regarding the error mgmt, do we have a central error file / error codes for these situations?

Reply:
we don't, we plan to work on it. Harshil plans to start something in badger.

Reply:
Can you instead say something like `the first argument to eq should be len`? Would that be more helpful to our users?